Source/Core/VideoCommon/ShaderCache.h

```diff
@@ -108,12 +108,20 @@ class ShaderCache final
   void AppendGXPipelineUID(const GXPipelineUid& config);
 
   // ASync Compiler Methods
-  void QueueVertexShaderCompile(const VertexShaderUid& uid);
-  void QueueVertexUberShaderCompile(const UberShader::VertexShaderUid& uid);
-  void QueuePixelShaderCompile(const PixelShaderUid& uid);
-  void QueuePixelUberShaderCompile(const UberShader::PixelShaderUid& uid);
-  void QueuePipelineCompile(const GXPipelineUid& uid);
-  void QueueUberPipelineCompile(const GXUberPipelineUid& uid);
+  void QueueVertexShaderCompile(const VertexShaderUid& uid, u32 priority);
+  void QueueVertexUberShaderCompile(const UberShader::VertexShaderUid& uid, u32 priority);
+  void QueuePixelShaderCompile(const PixelShaderUid& uid, u32 priority);
+  void QueuePixelUberShaderCompile(const UberShader::PixelShaderUid& uid, u32 priority);
+  void QueuePipelineCompile(const GXPipelineUid& uid, u32 priority);
+  void QueueUberPipelineCompile(const GXUberPipelineUid& uid, u32 priority);
+
+  // Priorities for compiling. The lower the value, the sooner the pipeline is compiled.
+  // The shader cache is compiled last, as it is the least likely to be required. On demand
+  // shaders are always compiled before pending ubershaders, as we want to use the ubershader
+  // for as few frames as possible, otherwise we risk framerate drops.
+  const u32 COMPILE_PRIORITY_ONDEMAND_PIPELINE = 100;
```

Comment:
Wouldn't it be better to use an enum class for this sort of thing?

Reply:
That was my initial thought, but casting all the values to u32 is messy, and any arbitrary number is valid. Unless I'm missing something with enum classes?

Reply:
Wouldn't it be sufficient to just change the parameters/types to some hypothetical

```cpp
enum class CompilePrority
```

as well, given no arithmetic is done on the parameters (as far as I can tell). Is using arbitrary numbers also a good idea as opposed to explicitly stating what priorities exist/are used in one spot? (don't know if any future work would be using that quality or not, which is why I ask). If it is, these could probably just be put into a nameless enum or be `static constexpr` members

Reply:
The reason behind arbitrary numbers is so we can expand the priorities later on. For example, priorities 0-10 for title screen, 10-20 for early game shaders, 20-30 for mid-game, etc. These may be based dynamically on elapsed time.

A nameless enum does sound like a better idea, though.